feat(faq): adding faq docs around using the csi driver (#763)
kiali/kiali.io: content/en/docs/FAQ/installation.md

kiali-56bbfd644-nkhlw   2/2     Running   0          43s
```

> ### How can I specify a container image digest hash when installing Kiali Server and Kiali Operator?

Author: I just changed the case to match the existing questions

helm install --set image.tag=7336eb77199a4d737435a8bf395e1666b7085cc7f0ad8b4cf9456b7649b7d6ad --set image.digest=sha256 ...and the rest of the helm install options...
```

### How can I use a CSI Driver to expose a custom secret to the Kiali Server?

> We will assume that you have a [CSI driver and provider installed](https://secrets-store-csi-driver.sigs.k8s.io/introduction) 

Reviewer: ```suggestion
You first must already have a [CSI driver and provider installed](https://secrets-store-csi-driver.sigs.k8s.io/introduction) 
```

in your cluster and a valid [SecretProviderClass](https://secrets-store-csi-driver.sigs.k8s.io/concepts.html?#secretproviderclass) deployed in the namespace where Kiali is installed. 

To mount a secret exposed by the CSI Driver, you can use the [custom_secret](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.deployment.custom_secrets) configuration 

> to supply the [CSI volume source](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume/#Volume) on the pod. The custom_secrets docs have an example. 

Reviewer: ```suggestion
to supply the [CSI volume source](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume/#Volume) on the pod. The [Kiali CR reference docs](https://kiali.io/docs/configuration/kialis.kiali.io/#example-cr) have an example. 
```

> The helm chart will automatically expose the secret as a volume mount into the container at the specified location.

Reviewer: ```suggestion
The Kiali Operator or server helm chart will automatically expose the secret as a volume mount into the container at the specified mount location.
```

Although Kiali retrieves the secret over the Kubernetes API, [mounting the secret](https://secrets-store-csi-driver.sigs.k8s.io/topics/sync-as-kubernetes-secret) is required for the CSI Driver to create the backing Kubernetes secret. 

> 

Reviewer: This would also be a good place to put a note about the "optional" flag being ignored when mounting CSI secrets. So let's get that in here.

```suggestion
Note that the [`custom_secrets` `optional` flag](https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.deployment.custom_secrets[*].optional) is ignored when mounting secrets from the CSI provider. The secrets are required to exist - then cannot be optional.

```

Author: Good catch!
